src/chainstate/burn/operations/delegate_stx.rs

```diff
@@ -680,5 +794,68 @@ mod tests {
         assert_eq!(op.delegated_ustx, u128::from_be_bytes([1; 16]));
         assert_eq!(op.delegate_to, StacksAddress::new(22, Hash160([2u8; 20])));
         assert_eq!(op.until_burn_height, Some(u64::from_be_bytes([1; 8])));
+        assert_eq!(op.memo, vec![1; 27]);
+    }
+
+    // Parse a DelegateStx op in which data is longer than 77, so that it is truncated.
```

Comment:
Is this desirable behavior?  IIRC OP_RETURNs can be as long as 520 bytes, but the relay policy constrains them to 80 bytes.  Miners can directly include 520-byte OP_RETURNs, though.  Do we want to support this?

Reply:
My reasoning here is as follows: 
- TransferStxOp does already parse a memo field, and the parsing essentially fails if there are more than 80 bytes: 
```
if data.len() > (61 + 16) {
            // too long
            warn!(
                "TransferStxOp payload is malformed ({} bytes, expected <= {})",
                data.len(),
                16 + 61
            );
            return None;
        }
```
- I wanted to model the parsing of StackStx and DelegateStx off of TransferStx, but I couldn't "fail" the parsing if there were more than 80 bytes (because that it not backwards compatible), so that's why I thought truncating it would mirror that behavior while maintaining backwards compatibility.

Reply:
If we're going to support OP_RETURN payloads greater than 80 bytes, then we'll need to make sure the burnchain parser logic will accept this.